### DISCUSSION TITLE
Nodemailer working but "Gmail couldn't verify that domain actually sent this message" error

### DISCUSSION THREAD

**Discussion by csjui** [vercel/community#4487] [Help] [answered]
### Summary

I've set up nodemailer in vercel using the provided vercel.app domain. When I set this domain as the from domain in nodemailer, gmail gives the warning "Gmail couldn't verify that domain actually sent this message". I've read that it has something to do with the SPF TXT record, but am unsure if it's possible to change this on the provided vercel domain? I currently have the same nodemailer app running on heroku with its basic domain, and it doesn't cause this error in gmail.

Any tips much appreciated!

### Steps to Reproduce

See above

**Comment by amyegan** [comment]
Because of the dynamic nature of Vercel deployments, many things behave differently from a more traditional server setup. When using Nodemailer, we recommend using an external `host` from a third-party provider instead of the outbound SMTP connection from a Serverless Function.

https://vercel.com/guides/sending-emails-from-an-application-on-vercel

**Reply by csjui** [reply]
I am using an external SMTP provider.

**Reply by csjui** [reply, marked as answer]
I fixed it. The domain my host had given me wasn't the actual domain being used to send the emails (probably just an alias). I found what that domain is and set it as the nodemailer "from" field and the warning not longer appears in gmail.
